### PR TITLE
feat: Intent bundles

### PR DESCRIPTION
**Pull request by taco-paco**
## Summary
This PR introduces Intent Bundles. It gives an ability for User to schedule multiple Intents. For example, Commit set of accounts `a`, and undelegate set `b`

## Compatibility
- [ ] No breaking changes
- [ ] Config change (describe):
- [ ] Migration needed (describe):

## Testing
- [x] tests (or explain)

## Summary by CodeRabbit

* **New Features**
  * Introduces an "intent bundle" workflow and new scheduling instruction to submit grouped commit, undelegate, and standalone actions.

* **Refactor**
  * Migrates scheduling, execution, persistence, DB, and service surfaces to bundle-centric APIs and messaging.

* **Bug Fixes**
  * Deduplicates committed-account entries during scheduling to prevent duplicate commits.

* **Tests**
  * Adds and updates unit and integration tests covering bundle scenarios, mixed intents, and undelegate edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: magicblock-committor-service/src/persist/commit_persister.rs [hunk 1]
```diff
@@ -509,42 +557,82 @@ mod tests {
...
 
     #[test]
-    fn test_start_base_message() {
+    fn test_create_commit_rows_undelegate_only() {
+        let message = create_test_message(1, undelegate_only_bundle());
+        let rows = IntentPersisterImpl::create_commit_rows(&message);
+
+        assert_eq!(rows.len(), 2);
+        assert!(rows.iter().all(|r| !r.undelegate));
+    }
```
[the remark this answers is not in the document]
Comment (taco-paco) [reply magicblock-committor-service/src/persist/commit_persister.rs]: True, done

File: programs/magicblock/src/magic_scheduled_base_intent.rs [hunk 2]
```diff
@@ -56,59 +57,94 @@ impl<'a, 'ic> ConstructionContext<'a, 'ic> {
...
     pub slot: Slot,
     pub blockhash: Hash,
-    pub action_sent_transaction: Transaction,
+    pub intent_bundle_sent_transaction: Transaction,
     pub payer: Pubkey,
-    // Scheduled action
-    pub base_intent: MagicBaseIntent,
+    /// Scheduled intent bundle
+    // TODO(edwin): rename
+    pub intent_bundle: MagicIntentBundle,
```
[the remark this answers is not in the document]
Comment (taco-paco) [reply programs/magicblock/src/magic_scheduled_base_intent.rs]: Removed as done

File: test-integration/programs/flexi-counter/src/processor/schedule_intent.rs [hunk 3]
```diff
@@ -138,3 +139,189 @@ pub fn process_create_intent(
...
+                }
+            })
+            .collect::<Vec<_>>();
+
+        // Post-undelegate actions
+        let undelegate_handlers = undelegate_counters
+            .iter()
+            .zip(undelegate_escrows.iter().cloned())
+            .zip(counter_diffs.iter().copied())
+            .map(|((counter, escrow_authority), counter_diff)| {
```
[the remark this answers is not in the document]
Comment (taco-paco) [reply test-integration/programs/flexi-counter/src/processor/schedule_intent.rs]: Fine for test

File: test-integration/programs/flexi-counter/src/processor.rs [hunk 4]
```diff
@@ -94,6 +96,18 @@ pub fn process(
...
+            num_undelegate,
+            counter_diffs,
+            compute_units,
+        } => process_create_intent_bundle(
+            accounts,
+            num_commit_only,
+            num_undelegate,
+            counter_diffs,
+            compute_units,
+        ),
```
[the remark this answers is not in the document]
Comment (taco-paco) [reply test-integration/programs/flexi-counter/src/processor.rs]: Extra for this smart contract

File: magicblock-committor-service/src/intent_execution_manager/intent_execution_engine.rs [hunk 5]
```diff
@@ -240,18 +234,18 @@ where
...
     }
 
-    /// Returns [`ScheduledBaseIntentWrapper`] from external channel
+    /// Returns [`ScheduleIntentBundleWrapper`] from external channel
     async fn get_new_intent(
-        receiver: &mut mpsc::Receiver<ScheduledBaseIntentWrapper>,
+        receiver: &mut mpsc::Receiver<ScheduledIntentBundle>,
         db: &Arc<D>,
-    ) -> Result<ScheduledBaseIntentWrapper, IntentExecutionManagerError> {
+    ) -> Result<ScheduledIntentBundle, IntentExecutionManagerError> {
```
[the remark this answers is not in the document]
Comment (taco-paco) [reply magicblock-committor-service/src/intent_execution_manager/intent_execution_engine.rs]: Done
